File: solutions-by-language/java/3005_count_elements_with_maximum_frequency.java
import java.util.*;

class Solution {
    public int maxFrequencyElements(int[] nums) {
        Map<Integer, Integer> freq = new HashMap<>();
        int maxFreq = 0, total = 0;

        for (int num : nums) {
            int count = freq.getOrDefault(num, 0) + 1;
            freq.put(num, count);
            maxFreq = Math.max(maxFreq, count);
        }

        for (int count : freq.values()) {
            if (count == maxFreq) total += count;
        }

        return total;
    }
}

File: solutions-by-language/cpp/3005_count_elements_with_maximum_frequency.cpp
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> freq;
        int maxFreq = 0, total = 0;

        for (int num : nums) {
            freq[num]++;
            maxFreq = max(maxFreq, freq[num]);
        }

        for (auto& [_, count] : freq) {
            if (count == maxFreq) total += count;
        }

        return total;
    }
};